lib/Adapter/TolerantParser/Fixer/DocblockIndentationFixer.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+namespace Phpactor\CodeBuilder\Adapter\TolerantParser\Fixer;
+
+use Microsoft\PhpParser\Node;
+use Microsoft\PhpParser\Node\MethodDeclaration;
+use Microsoft\PhpParser\Node\PropertyDeclaration;
+use Microsoft\PhpParser\Node\SourceFileNode;
+use Microsoft\PhpParser\Node\Statement\ClassDeclaration;
+use Microsoft\PhpParser\Parser;
+use Phpactor\CodeBuilder\Domain\TextEdit;
+use Phpactor\CodeBuilder\Domain\StyleProposer;
+use Phpactor\CodeBuilder\Domain\TextEdits;
+use Phpactor\CodeBuilder\Util\TextFormat;
+use Phpactor\CodeBuilder\Util\TextUtil;
+
+class DocblockIndentationFixer implements StyleProposer
+{
+    /**
+     * @var Parser
+     */
+    private $parser;
+
+    /**
+     * @var TextFormat
+     */
+    private $textFormat;
+
+    public function __construct(Parser $parser, ?TextFormat $textFormat = null)
+    {
+        $this->parser = $parser;
+        $this->textFormat = $textFormat ?: new TextFormat();
+    }
+
+    public function propose(string $text): TextEdits
+    {
+        $edits = [];
+        $node = $this->parser->parseSourceFile($text);
+
+        foreach ($this->docblockNodes($node) as $docblockNode) {
+            $edits[] = $this->indentationEdits($docblockNode);
+        }
+
+        return TextEdits::fromTextEdits($edits);
+    }
+
+    private function indentationEdits(Node $node): TextEdit
+    {
+        $newLines = [];
+        $baseIndent = '';
+        $lines = TextUtil::lines($node->getLeadingCommentAndWhitespaceText());
+
+        foreach ($lines as $line) {
+            if (preg_match('{^\s*\*}', $line)) {
+                $line = $this->textFormat->indentRemove($line);
+                $line = $baseIndent .' '. $line;
+            }
+
+            if (preg_match('{^\s*/\*\*}', $line)) {
+                $baseIndent = TextUtil::lineIndentation($line);
+            }
+
+            $newLines[] = $line;
+        }
+
+        return new TextEdit(
+            $node->getFullStart(),
+            $node->getStart() - $node->getFullStart(),
+            $this->textFormat->implodeLines($newLines)
```

Comment:
Can you also add a test for properties with phpdoc ?
Without phpdoc there must be no empty line between them, with there must be one.
It will be a tricky one to handle, I would say that there must always be a new line before a docblock:
```suggestion
            $this->textFormat->newLineChar() . $this->textFormat->implodeLines($newLines)
```

At the exception of the first node after the class opening brace...

Reply:
We should probably add it in the MemberEmptyLineFixer

Reply:
Updated. It's handled in the MemberEmptyLineFixer